lib/src/exceptions.dart

```diff
@@ -35,7 +35,7 @@ class FormatterException implements Exception {
         source += " " * (error.offset + error.length - source.length);
       }
 
-      var file = new SourceFile(source, url: error.source.fullName);
+      var file = new SourceFile.fromString(source, url: error.source.fullName);
```

Comment:
Is this because of a change in the analyzer API? If so, you'll need to increase the minimum version constraint on analyzer.